assets/js/src/segments/dynamic/dynamic_segments_filters/woocommerce.tsx

```diff
@@ -70,21 +85,67 @@ export const WooCommerceFields: React.FunctionComponent<Props> = ({ onChange, it
         </div>
       </div>
     );
-  }
+  } else if (item.action === WooCommerceActionTypes.PURCHASED_CATEGORY) {
+    optionFields = (
+      <div className="mailpoet-form-field">
+        <div className="mailpoet-form-input mailpoet-form-select">
+          <Select
+            placeholder={MailPoet.I18n.t('selectWooPurchasedCategory')}
+            options={categoryOptions}
+            value={find(['value', item.category_id], categoryOptions)}
+            onChange={(option: SelectOption): void => compose([
```

Comment:
I copied this `onChange` from the other action types but I don't fully understand what is happening here.